splot/_viz_bokeh.py

```diff
@@ -0,0 +1,453 @@
+""" 
+Leightweight interactive visualizations in Bokeh.
+
+TODO: 
+We are re-projection data into web-mercator atm, to allow plotting from raw coordinates.
+The user should however be aware of the projection of his data.
+"""
+
+__author__ = ("Stefanie Lumnitz <stefanie.lumitz@gmail.com>")
+
+import pandas as pd
+import pysal as ps
+import esda
+from bokeh.plotting import figure
+from bokeh.models import (GeoJSONDataSource, ColumnDataSource,
+                          CategoricalColorMapper, Span,
+                          HoverTool, Legend)
+from bokeh.layouts import gridplot
+from bokeh import palettes 
+
+from ._viz_utils import (bin_labels_choropleth, add_legend,
+                         mask_local_auto, calc_data_aspect)
+
+
+def plot_choropleth(df, attribute, title=None, plot_width=500,
+                    plot_height=500, method='quantiles',
+                    k=5, reverse_colors=False, tools='', region_column=''):
+    '''
+    Plot Choropleth colored according to attribute
+    
+    Parameters
+    ----------
+    df : Geopandas dataframe
+        Dataframe containign relevant shapes and attribute values.
+    attribute : str
+        Name of column containing attribute values of interest.
+    title : str, optional
+        Title of map. Default title=None
+    plot_width : int, optional
+        Width dimension of the figure in screen units/ pixels.
+        Default = 500
+    plot_height : int, optional
+        Height dimension of the figure in screen units/ pixels.
+        Default = 500
+    method : str, optional
+        Classification method to be used. Options supported:
+        * 'quantiles' (default)
+        * 'fisher-jenks'
+        * 'equal-interval'
+    k : int, optional
+        Number of bins, assigning values to. Default k=5
+    reverse_colors: boolean
+        Reverses the color palette to show lightest colors for
+        lowest values. Default reverse_colors=False
+    tools : str, optional
+        Tools used for bokeh plotting. Default = ''
+    region_column : str, optional
+        Column name containing region descpriptions/ names or polygone ids.
+        Default = ''.
+    
+    Returns
+    -------
+    fig : Bokeh Figure instance
+        Figure of Choropleth
+    
+    Examples
+    --------
+    >>> import libpysal.api as lp
+    >>> from libpysal import examples
+    >>> import geopandas as gpd
+    >>> import esda
+    >>> from splot.bk import plot_choropleth
+    >>> from bokeh.io import show
+
+    >>> link = examples.get_path('columbus.shp')
+    >>> df = gpd.read_file(link)
+    >>> w = lp.Queen.from_dataframe(df)
+    >>> w.transform = 'r'
+
+    >>> TOOLS = "tap,help"
+    >>> fig = plot_choropleth(df, 'HOVAL', title='columbus',
+    ...                       reverse_colors=True, tools=TOOLS)
+    >>> show(fig) 
+    '''
+    # We're adding columns, do that on a copy rather than on the users' input
+    df = df.copy()
```

Comment:
I thought from our last conversation, we agreed that `df.assign(...)` would make a copy already? In a few cases, you're making more than one new column, so keep this copying as is I guess. 

But, if I recall correctly, @jorisvandenbossche was suggesting like... for the column name-style argument in  `geodataframe.plot('colname', **options)`, you could pass `geodataframe.plot(numpy_vector, **options)` to plot without `assign`, or `copy`?

Reply:
Ok, yep for the `_viz_mpl.py` file I am using the df.assign() already. My thoughts were, too, that df.assign might not work if I have to add multiple columns which is the case for the three-plot visualisations. 

However, I am not using the .plot() functionality in `_viz_bokeh.py` at all and was wondering if you guys would have any suggestions how to handle adding columns here, besides using "df.copy()". I am happy to also just leave this as it is for now and add a note for future improvements.